### PR TITLE
refactor raftstore

### PR DESCRIPTION
**Pull request by coocood**
- extract GlobalContext for immutable singletons.
- Separate RaftContext and StoreContext.
- Make names more consistent with their behaviors.

### CODE REVIEW

**Comment by coocood** [comment]
@cfzjywxk @bobotu PTAL

File: tikv/raftstore/fsm_peer.go [hunk 1]
```diff
@@ -93,10 +93,18 @@ func (pf *peerFsm) regionID() uint64 {
...
 
+func (d *peerFsm) region() *metapb.Region {
+	return d.peer.Store().region
+}
+
 func (pf *peerFsm) getPeer() *Peer {
 	return pf.peer
 }
 
+func (d *peerFsm) storeID() uint64 {
```
Comment (cfzjywxk) [inline tikv/raftstore/fsm_peer.go]: Make the receiver variable name all the same?

File: tikv/raftstore/fsm_store.go [hunk 2]
```diff
@@ -461,78 +375,75 @@ func (bs *raftBatchSystem) start(
...
+		snapMgr:              snapMgr,
 		router:               bs.router,
-		splitCheckScheduler:  workers.splitCheckWorker.scheduler,
-		regionScheduler:      workers.regionWorker.scheduler,
-		raftLogGCScheduler:   workers.raftLogGCWorker.scheduler,
-		compactScheduler:     workers.compactWorker.scheduler,
-		pdScheduler:          workers.pdWorker.scheduler,
-		computeHashScheduler: workers.computeHashWorker.scheduler,
 		trans:                trans,
+		pdScheduler:          bs.workers.pdWorker.scheduler,
```
Comment (cfzjywxk) [inline tikv/raftstore/fsm_store.go]: Should we remove the `workers` fields and save each worker in `GlobalContext`

File: tikv/raftstore/fsm_store.go [hunk 3]
```diff
@@ -62,37 +62,44 @@ func (m *storeMeta) setRegion(region *metapb.Region, peer *Peer) {
...
-	tickDriverCh         chan<- uint64
 	trans                Transport
-	queuedSnaps          map[uint64]struct{}
 	pdScheduler          chan<- task
-	raftLogGCScheduler   chan<- task
-	store                *metapb.Store
 	regionScheduler      chan<- task
-	splitCheckScheduler  chan<- task
 	computeHashScheduler chan<- task
+	raftLogGCScheduler   chan<- task
```
Comment (cfzjywxk) [inline tikv/raftstore/fsm_store.go]: Should we use names like `xxxTaskQueue`  and `xxxTaskHandler/Processor`  in unistore

File: tikv/raftstore/fsm_store.go [hunk 4]
```diff
@@ -62,37 +62,44 @@ func (m *storeMeta) setRegion(region *metapb.Region, peer *Peer) {
 type mergeLock struct {
 }
 
-type PollContext struct {
+type GlobalContext struct {
```
Comment (cfzjywxk) [inline tikv/raftstore/fsm_store.go]: Maybe we could do more split on this?  `Global` still seems not quite direct :eyes:

File: tikv/raftstore/fsm_store.go (hunk 2 above)
Comment (coocood) [reply tikv/raftstore/fsm_store.go]: workers are not used after start, we don't need them in `GlobalContext`.

File: tikv/raftstore/fsm_store.go (hunk 4 above)
Comment (coocood) [reply tikv/raftstore/fsm_store.go]: split by what?

**Comment by cfzjywxk** [comment]
LGTM maybe we need to add some unit test cases for `raftStore` related code paths

**Pull request merged**